lib/arjdbc/mysql/adapter.rb

```diff
@@ -40,6 +40,12 @@ def initialize(connection, logger, connection_parameters, config)
         # configure_connection taken care of at ArJdbc::Abstract::Core
       end
 
+      def self.database_exists?(config)
+        !!ActiveRecord::Base.mysql2_connection(config)
```

Comment:
not sure about this pattern, if we'd like to support JNDI ... connection might be lazy.
isn't there an existing API we could use or maybe do a conn establish + explicit ping?

Reply:
Nice catch. Haven't thought about that, just blindly copied AR. And thinking about it, this is also a possible connection leak. Will fix up in the next few days

Reply:
Fixed. Now the check is forced, regardless of JNDI and the connection is closed afterwards